betterproto/tests/inputs/oneof_default_value_serialization/test_oneof_default_value_serialization.py

```diff
@@ -0,0 +1,74 @@
+import pytest
+import datetime
+
+import betterproto
+from betterproto.tests.output_betterproto.oneof_default_value_serialization import (
+    Test,
+    Message,
+    NestedMessage,
+)
+
+
+def assert_round_trip_serialization_works(message: Test) -> None:
+    assert betterproto.which_one_of(message, "value_type") == betterproto.which_one_of(
+        Test().from_json(message.to_json()), "value_type"
+    )
+
+
+def test_oneof_default_value_serialization_works_for_all_values():
+    """
+    Serialization from message with oneof set to default -> JSON -> message should keep
+    default value field intact.
+    """
+
+    test_cases = [
+        Test(bool_value=False),
+        Test(int64_value=0),
+        Test(
+            timestamp_value=datetime.datetime(
+                year=1970,
+                month=1,
+                day=1,
+                hour=0,
+                minute=0,
+                tzinfo=datetime.timezone.utc,
+            )
+        ),
+        Test(duration_value=datetime.timedelta(0)),
+        Test(wrapped_message_value=Message(value=0)),
+        # NOTE: Do NOT use betterproto.BoolValue here, it will cause JSON serialization
+        # errors.
+        # TODO: Do we want to allow use of BoolValue directly within a wrapped field or
+        # should we simply hard fail here?
+        Test(wrapped_bool_value=False),
```

Comment:
Good point!

It seems odd that passing a `BoolValue` to a field that is defined as such will break the library.
Of course, it was meant as a convenience that the BoolValue is not required, but breaking it goes a bit far.

It does raise some questions.
Do we always store a `bool` internally? Then what happens if you store a BoolValue, then read it again.. it becomes a `bool`. That may be confusing, and potentially cause bugs.

We may need to look into this, in a separate issue.

Its fine with me to keep the TODO

Reply:
Yeah if I had to make a choice (with just thinking about this briefly) I would err on the side of failing fast when attempting to assign a non-primitive to a wrapped field. That at least makes the usage of the field unambiguous